src/array/random.jl

```diff
@@ -0,0 +1,30 @@
+using Random
+
+Random.rand!(dm::DArray{T}) where T = map!(_ -> rand(T), dm)
+Random.randn!(dm::DArray{T}) where T = map!(_ -> randn(T), dm)
+
+randfork(rng::R, n::Integer) where {R<:AbstractRNG} =
+    R(rand(copy(rng), Int) + n)
+
+function Random.rand!(rng::AbstractRNG, A::DArray{T}) where T
+    part_sz = prod(map(length, first(A.subdomains).indexes))
+    Dagger.spawn_datadeps() do
+        for Ac in chunks(A)
+            rng = randfork(rng, part_sz)
+            new_rng = copy(rng)
```

Comment:
Is it necessary to copy `rng` again here since `randfork()` creates a new one anyway? (same thing for the `randn!()` definition below)

Reply:
I had this here because we may pass the `rng` argument as-is to the task, and if it's executed on a local thread, we want to make sure that it won't alias with future uses of `rng` within this function. But datadeps tasks don't start until the end of the block, so it should be fine to not copy it again.